### PR TITLE
Feature/email service

### LINKED ISSUE #38: setup email service
some random guy made a [`nodemailer`](https://nodemailer.com/about/) service for feathers: https://github.com/feathersjs/feathers-mailer.

- [x] add new service to `notifications` topic, in the `services` folder
- [x] hook service up into the main `server` top-level file
- [x] install `maildev` as `devDependency`: https://github.com/djfarrelly/maildev
- [x] add `dev:mail` script, move current `dev` script to `dev:app`, change `dev` script to be `npm-run-all -p dev:*`
- [x] add necessary development `mailer` config for `maildev`: https://github.com/djfarrelly/maildev#configure-your-project

### PR DESCRIPTION
**Pull request by iainkirkpatrick**
a basic email service (using [feathers-mailer](https://github.com/feathersjs/feathers-mailer))

todo:
- [x] weird error with not grabbing the right email from credentials on create Agent hook, suspect this might be a bug introduced by myself / Juliet regarding renaming `contextAgent`

closes #38

### CODE REVIEW

**Comment by iainkirkpatrick** [comment]
discussed the error/s with @ahdinosaur, due to some changes needed in `dogstack-agents`- these can be done in a separate ticket.

File: notifications/services/mailer.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const feathersMailer = require('feathers-mailer')
+const nodemailer = require('nodemailer')
```
Comment (ahdinosaur) [inline notifications/services/mailer.js]: not necessary anymore

File: package.json [hunk 2]
```diff
@@ -4,7 +4,9 @@
   "description": "Helping people buy good food at good prices, together.",
   "scripts": {
     "start": "dog server",
-    "dev": "dog dev server",
+    "dev:app": "dog dev server",
```
Comment (ahdinosaur) [inline package.json]: `dev:server`?

File: notifications/services/mailer.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+module.exports = function () {
+  const app = this
+  const db = app.get('db')
```
Comment (ahdinosaur) [inline notifications/services/mailer.js]: not needed

**Review by ahdinosaur — APPROVED** [review]
woo 🎉

**Pull request merged**